pkg/platform/packet/packet.go

```diff
@@ -247,13 +248,31 @@ func (c *config) GetExpectedNodes() int {
 
 // Check cluster config is valid
 func (c *config) checkValidConfig() hcl.Diagnostics {
+	var diagnostics hcl.Diagnostics
+
 	if len(c.WorkerPools) == 0 {
-		err := &hcl.Diagnostic{
+		diagnostics = append(diagnostics, &hcl.Diagnostic{
 			Severity: hcl.DiagError,
 			Summary:  "At least one worker pool must be defined",
 			Detail:   "Make sure to define at least one worker pool block in your cluster block",
+		})
+	}
+
+	// Check that worker pools names are unique
+	dup := make(map[string]bool)
+	for _, w := range c.WorkerPools {
+		if !dup[w.Name] {
```

Comment:
```suggestion
		if _, ok :=dup[w.Name]; !ok {
```

Reply:
Hm, maybe boolean type actually make sense then, if this makes this condition easier. Seems "smart" though :smile:

Reply:
I'll leave it as boolean